deck-builder/main.js

```diff
@@ -422,6 +1139,71 @@ function clearList() {
   }
 }
 
+function addCard(card) {
+  if ($deckList.length < 50) {
```

Comment:
Try to flatten your nesting.
```js
function addCard(card) {
  if (deckList.length === 50) {
    return
  }
  // continute algorithm
}
```

This could probably be used throughout this function. If it helps cut back on loops, try massaging the data before running the algorithm.

```js
var allCards = []
for (var i = 0; i < boosterPacks.length; i++) {
  allCards = allCards.concat(boosterPacks[i].cards)
}
```